### PR TITLE
Add NCS36510 support

### PR DESCRIPTION
**Pull request by maclobdell**
Test Summary:
Pass count 15 of 15 tests
FLASH TEST SCRIPT PASSED
INFO:root:DAP SWD MODE initialised
INFO:root:6 hardware breakpoints, 4 literal comparators
INFO:root:CPU core is Cortex-M3
INFO:root:4 hardware watchpoints
INFO:root:Programmed 24576 bytes (12 pages) at 16.38 kB/s
INFO:root:DAP SWD MODE initialised
INFO:root:6 hardware breakpoints, 4 literal comparators
INFO:root:CPU core is Cortex-M3
INFO:root:4 hardware watchpoints
INFO:root:Telnet: server started on port 4444
INFO:root:GDB server started at port:3334

### CODE REVIEW

**Comment by c1728p9** [comment]
@mbed-bot: TEST

File: pyOCD/target/target_ncs36510.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+from cortex_m import CortexM
+from .memory_map import (FlashRegion, RamRegion, MemoryMap)
+import logging
+
+# NRF51 specific registers
+RESET = 0x40000544
+RESET_ENABLE = (1 << 0)
+
+class NCS36510(CortexM):
```
Comment (c1728p9) [inline pyOCD/target/target_ncs36510.py]: This should extend CoreSightTarget rather than CortexM.

File: pyOCD/target/target_ncs36510.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    memoryMap = MemoryMap(
+        FlashRegion(    start=0x2000,           length=0x50000,      blocksize=0x800, isBootMemory=True),
+        RamRegion(      start=0x3FFF4000,  length=0xC000)
+        )
+
+    def __init__(self, link):
+        super(NCS36510, self).__init__(link, self.memoryMap)
+
+    def resetn(self):
```
Comment (c1728p9) [inline pyOCD/target/target_ncs36510.py]: This doesn't do anything

File: test/flash_test.py [hunk 3]
```diff
@@ -127,7 +127,10 @@ def flash_test(board_id):
         if target_type == "nrf51":
             # Override clock since 10MHz is too fast
             test_clock = 1000000
-
+        if target_type == "ncs36510":
+            # Override clock since 10MHz is too fast
+            test_clock = 1000000
+
```
Comment (c1728p9) [inline test/flash_test.py]: Nit - whitespace

File: test/flash_test.py [hunk 4]
```diff
@@ -238,12 +241,12 @@ def print_progress(progress):
             print("TEST FAILED")
         test_count += 1
 
-        flash.setFlashAlgoDebug(False)
+        #flash.setFlashAlgoDebug(False)
```
Comment (c1728p9) [inline test/flash_test.py]: This should be kept on

File: test/speed_test.py [hunk 5]
```diff
@@ -83,7 +83,10 @@ def speed_test(board_id):
         if target_type == "nrf51":
             # Override clock since 10MHz is too fast
             test_clock = 1000000
-
+        if target_type == "ncs36510":
+            # Override clock since 10MHz is too fast
+            test_clock = 1000000
+
```
Comment (c1728p9) [inline test/speed_test.py]: nit - whitespace

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by c1728p9** [comment]
@mbed-bot: TEST

**Comment by maclobdell** [comment]
Everything still passing after you tweaked the commits.  Thanks!

Test Summary:
Pass count 15 of 15 tests
FLASH TEST SCRIPT PASSED
INFO:root:DAP SWD MODE initialised
INFO:root:6 hardware breakpoints, 4 literal comparators
INFO:root:CPU core is Cortex-M3
INFO:root:4 hardware watchpoints
INFO:root:Programmed 24576 bytes (12 pages) at 12.78 kB/s
INFO:root:DAP SWD MODE initialised
INFO:root:6 hardware breakpoints, 4 literal comparators
INFO:root:CPU core is Cortex-M3
INFO:root:4 hardware watchpoints
INFO:root:Telnet: server started on port 4444
INFO:root:GDB server started at port:3334

**Pull request merged**